ecs-agent/manageddaemon/managed_daemon.go

```diff
@@ -80,11 +82,60 @@ func NewManagedDaemon(
 	return newManagedDaemon
 }
 
+var ImportAll = defaultImportAll
+
 // ImportAll function will parse/validate all managed daemon definitions
 // defined in /var/lib/ecs/deps/daemons and will return an array
 // of valid ManagedDeamon objects
-func ImportAll() ([]*ManagedDaemon, error) {
-	return []*ManagedDaemon{}, nil
+func defaultImportAll() ([]*ManagedDaemon, error) {
```

Comment:
Just curious, is there a specific reasoning why we're using a var to refer to the this function instead of declaring it to be global and calling it directly?

Reply:
yes -- this is a pattern to allow us to override it in our testing (so we don't end up actually importing this unless we want to) see the agent/app/agent_test.go for where we override it with a dummy function.